pkg/adapter/kubernetes/restartcr/doguRestartRepo_test.go

```diff
@@ -0,0 +1,77 @@
+package restartcr
+
+import (
+	"context"
+	"errors"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	v1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func Test_doguRestartRepository_RestartAll(t *testing.T) {
+	t.Run("no error on restart all", func(t *testing.T) {
+		// given
+		testContext := context.Background()
+		testDoguSimpleName := common.SimpleDoguName("testdogu")
+		dogusThatNeedARestart := []common.SimpleDoguName{testDoguSimpleName}
+		mockDoguRestartInterface := NewMockDoguRestartInterface(t)
+
+		mockDoguRestartInterface.EXPECT().Create(testContext, mock.Anything, metav1.CreateOptions{}).Return(&v1.DoguRestart{
```

Comment:
If the `GenerateName` field works you can expect the explizit resource in the mock call.